models/record.go

```diff
@@ -351,7 +347,7 @@ func (r *Record) handlePayerInformation() []HasEntry {
 // TODO: this belongs on CodeSet.
 func reasonInCodes(code CodeSet, reason CodedConcept) bool {
 	for _, value := range code.Values {
-		if reason.Code == value.Code && reason.CodeSystem == value.CodeSystem {
+		if reason.Code == value.Code && reason.CodeSystem == value.CodeSystemName {
```

Comment:
@mattrbianchi 
Hi, I want to ask the reason for this change, I understand that these fields couldn't be same, for example I have concepts: 
`"concepts": [
    {
      "_id": "5bd13af6e0771700088868e0",
      "black_list": false,
      "code": "183932001",
      "code_system": "2.16.840.1.113883.6.96",
      "code_system_name": "SNOMED-CT",
      "code_system_version": "2017-03",
      "display_name": "Procedure contraindicated (situation)",
      "white_list": false
    }]`

We can see that the fields codeSystem and CodeSystemName  has own values and these aren't the same

Could you explain  the reason the change? Thanks for your help